File: BlogServer/Blog.Web/Controllers/Api/UserController.cs
using Blog.Model.Param;
using Blog.Model.Rsult;
using Blog.Service.Api.Interface;
using Blog.Utils;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Blog.Web.Controllers.Api
{
    [Route("api/user")]
    [ApiController]
    [Authorize]
    public class UserController : ControllerBase
    {
        public UserController(IUserService userService)
        {
            UserService = userService;
        }

        public IUserService UserService { get; }

        [HttpPost("insert")]
        public async Task<ResultStruct> Insert([FromBody] UserInsertParam param)
        {
            return await ResultFun.AsyncReturn(param, UserService.Insert);
        }

        [HttpPost("list")]
        public async Task<ResultStruct<PageResult<UserListRsult>>> List([FromBody] UserFindParam param)
        {
            return await ResultFun.AsyncReturn(param, UserService.List);
        }

        [HttpPost("update")]
        public async Task<ResultStruct> Update([FromBody] UserUpdateParam param)
        {
            return await ResultFun.AsyncReturn(param, UserService.Update);
        }

        [HttpPost("updatePassword")]
        public async Task<ResultStruct> UpdatePassword([FromBody] UserUpdatePasswordParam param)
        {
            return await ResultFun.AsyncReturn(param, UserService.UpdatePassword);
        }

        [HttpPost("updateUserPower")]
        public async Task<ResultStruct> UpdateUserPower([FromBody] UserUpdatePowerParam param)
        {
            return await ResultFun.AsyncReturn(param, UserService.UpdateUserPower);
        }

        [HttpPost("getUserPower")]
        public async Task<ResultStruct<int[]>> GetUserPower([FromBody] IDParam param)
        {
            return await ResultFun.AsyncReturn(param, UserService.GetUserPower);
        }

        [HttpDelete("delete")]
        public async Task<ResultStruct> Delete([FromBody] IDParam param)
        {
            return await ResultFun.AsyncReturn(param, UserService.Delete);
        }
    }
}

File: blog-server/src/module/admin/user/User.controller.ts
import {Controller, Post, Delete, Body, HttpStatus} from '@nestjs/common'
import { UserService } from './User.service'
import {ApiTags, ApiOperation, ApiResponse, ApiBody} from '@nestjs/swagger';
import {
    UserFindParam,
    UserInsertParam,
    UserUpdateParam,
    UserUpdatePasswordParam,
    UserUpdatePowerParam
} from "./model/User.param";
import {IdParam} from "../../../model/BaseParam";
import {PageRsult} from "../../../model/BaseRsult";
import {User} from "../../../db/entities/User";


@ApiTags('user')  // 用于给路由分组
@Controller('api/user')
export class UserController {
    constructor (private readonly userService: UserService) {}

    @ApiBody({ type: UserInsertParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, description: '成功'})
    @Post('insert')
    insert (@Body() insetBody: UserInsertParam) :Promise<void> {
        return this.userService.insert(insetBody)
    }

    @ApiBody({ type: UserFindParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, type: Array<User>, description: '成功'})
    @Post('list')
    list (@Body() listBody: UserFindParam): Promise<PageRsult<User>> {
        return this.userService.list(listBody)
    }

    @ApiBody({ type: UserUpdateParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, description: '成功'})
    @Post('update')
    update (@Body() updateBody: UserUpdateParam): Promise<void> {
        return this.userService.update(updateBody)
    }

    @ApiBody({ type: UserUpdatePasswordParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, description: '成功'})
    @Post('updatePassword')
    updatePassword (@Body() pwBody: UserUpdatePasswordParam): Promise<void> {
        return this.userService.updatePassword(pwBody)
    }

    @ApiBody({ type: UserUpdatePowerParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, description: '成功'})
    @Post('updateUserPower')
    updateUserPower (@Body() body: UserUpdatePowerParam): Promise<void> {
        return this.userService.updateUserPower(body)
    }

    @ApiBody({ type: IdParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, type: Array<number>, description: '成功'})
    @Post('getUserPower')
    getUserPower (@Body() getBody: IdParam): Promise<Array<number>> {
        return this.userService.getUserPower(getBody)
    }

    @ApiBody({ type: IdParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, description: '成功'})
    @Delete('delete')
    delete (@Body() deleteBody: IdParam): Promise<void> {
        return this.userService.delete(deleteBody)
    }
}